src/toil/job.py

```diff
@@ -1288,9 +1288,10 @@ def deleteLocalFile(self, fileStoreID):
                                     logger.debug(fileToDelete,  'was read mutably and deleted by '
                                                  'the user')
                                 else:
-                                    raise CacheError('Illegal operation detected. Cache tracked'
-                                                     'file deleted explicitly by user. Use'
-                                                     'deleteLocalFile to delete such files.')
+                                    raise CacheIllegalDeleteError('Cache tracked file deleted '
```

Comment:
You might want to include the offending file name in the message. But I am not sure.